### ISSUE TITLE
💅 [useNamingConvention] Ignore declarations inside an external TypeScript module

### ISSUE THREAD

**Issue by cr7pt0gr4ph7** [biomejs/biome#4545]
### Rule name

`lint/style/useNamingConvention`

### Description

The documentation says that "declarations inside an external TypeScript module" should be ignored by `useNamingConvention` (as described at the bottom of the [relevant docs section](https://biomejs.dev/linter/rules/use-naming-convention/#ignored-declarations)), but the implementation actually complains about them.

#### Example:
```ts
declare module "myExternalModule" {
  export interface my_INTERFACE {}
}
```

#### Expected Result
 No Errors are reported

#### Actual Result
Lint is generated for `my_INTERFACE`.

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=ZABlAGMAbABhAHIAZQAgAG0AbwBkAHUAbABlACAAIgBtAHkARQB4AHQAZQByAG4AYQBsAE0AbwBkAHUAbABlACIAIAB7AAoAIAAgAGUAeABwAG8AcgB0ACAAaQBuAHQAZQByAGYAYQBjAGUAIABtAHkAXwBJAE4AVABFAFIARgBBAEMARQAgAHsAfQAKAH0ACgA%3D

### Environment information

https://github.com/biomejs/biome/actions/runs/11854065273/job/33035446404?pr=4542

### Tangentially related issues:
- #2171

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
